extends Control


func _ready() -> void:
	# 1. Conectar la se침al `dialogue_ended` a una funci칩n que t칰 definir치s.
	DialogueManager.dialogue_ended.connect(_on_dialogue_ended)
	
	# 2. Iniciar el di치logo (como ya lo tienes)
	DialogueManager.show_example_dialogue_balloon(preload("res://dialogues/historia.dialogue"))

# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta: float) -> void:
	pass

# 游뚿 Esta funci칩n se llamar치 autom치ticamente cuando el di치logo termine
func _on_dialogue_ended(dialogue):
	# Es buena pr치ctica esperar un momento para asegurar que el DialogueManager termine la limpieza
	await get_tree().create_timer(0.2).timeout
	
	# Opcional: Si estabas usando el GameManager, desact칤valo aqu칤
	# GameManager.activo = false 
	
	# 3. Cambiar a la escena final
	get_tree().change_scene_to_file("res://Scenes/principio.tscn")
